### DISCUSSION TITLE
CUE loop output

### DISCUSSION THREAD

**Discussion by tdenad16** [cue-lang/cue#2114] [Q&A] [answered]
Hi! I have just started learning CUE and noticed that CUE output in loops are inconsistent or rather no patterns as to how it is evaluated.

example is from a search result on the website:
https://cuelang.org/play/?id=FkUiTP33ZyM#cue@export@cue

### **Example:**

```
ex0: {
    aardvark: "oink"
    yyy: "why why why"
    for x,a in [1,2] {
        for y,b in ["A","B"] {
            "\(a)\(b)":"\(x),\(y)"
        }
        "after_\(a)": "hi"
    }
    zzzz: "else"
}

ex1: {
    for x,a in [1,2] {
        "loop_\(a)": "hi"
    }
    zis_shouldbe: "at the end"
}

ex2: {
    a_thisis: "at the top"
    for x,a in [1,2] {
        "loop_\(a)": "hi"
    }
    z_shouldbe: "at the end"
}
```

### **Example output in CUE Playground:**

```
ex0: {
	aardvark: "oink"
	yyy:      "why why why"
	after_1:  "hi"
	after_2:  "hi"
	"1A":     "0,0"
	"1B":     "0,1"
	"2A":     "1,0"
	zzzz:     "else"
	"2B":     "1,1"
}

ex1: {
	zis_shouldbe: "at the end"
	loop_1:       "hi"
	loop_2:       "hi"
}

ex2: {
	a_thisis:   "at the top"
	loop_1:     "hi"
	z_shouldbe: "at the end"
	loop_2:     "hi"
}
```

### **What I expect th output to be:**

```
ex0: {
	aardvark: "oink"
	yyy:      "why why why"
	"1A":     "0,0"
	"1B":     "0,1"
	after_1:  "hi"
	"2A":     "1,0"
	"2B":     "1,1"
	after_2:  "hi"
	zzzz:     "else"
}

ex1: {
	loop_1:       "hi"
	loop_2:       "hi"
	zis_shouldbe: "at the end"
}

ex2: {
	a_thisis:   "at the top"
	loop_1:     "hi"
	loop_2:     "hi"
	z_shouldbe: "at the end"
}
```

I can't seem to find discussions or topics where this was discussed. 
If there is and I have missed it, please redirect me or guide me to it.

Thank you!

**Comment by myitcv** [comment, marked as answer]
Hi @tdenad16 - thanks for asking. Right now, the order of iteration of structs, and the way they interact with comprehensions, is not specified. We plan to cover this in https://github.com/cue-lang/cue/issues/474.

**Reply by tdenad16** [reply]
@myitcv - Thank you so much for the clarification! I will be looking forward for the updates.
